csharp/src/Ice/Internal/Incoming.cs

```diff
@@ -233,20 +233,14 @@ public void invoke(ServantManager servantManager, Ice.InputStream stream)
     public Task<Ice.OutputStream> setResult(Ice.OutputStream os)
     {
         _os = os;
-        return null; // Response is cached in the Incoming to not have to create unecessary Task
+        return null; // Response is cached in the Incoming to not have to create unnecessary Task
     }
 
     public Task<Ice.OutputStream> setMarshaledResult<T>(T result) where T : Ice.MarshaledResult
     {
-        if (result == null)
```

Comment:
Not sure why we handled null results here. We had no test for this odd situation.